app/models/middleware_server.rb

```diff
@@ -5,4 +5,50 @@ class MiddlewareServer < ActiveRecord::Base
   has_many :middleware_deployments, :foreign_key => "server_id", :dependent => :destroy
   serialize :properties
   acts_as_miq_taggable
+
+  include LiveMetricsMixin
+
+  # Used to describe Hawkular supported metrics for this resource and convert it into MiQ style names
+  METRICS_HWK_MIQ = {
+      "WildFly Memory Metrics~Heap Used"                                  => "mw_heap_used",
+      "WildFly Aggregated Web Metrics~Aggregated Servlet Request Time"    => "mw_agregated_servlet_time",
+      "WildFly Memory Metrics~NonHeap Committed"                          => "mw_nonheap_comitted",
+      "WildFly Aggregated Web Metrics~Aggregated Expired Web Sessions"    => "mw_aggregated_expired_web_sessions",
+      "WildFly Aggregated Web Metrics~Aggregated Max Active Web Sessions" => "mw_aggregated_max_active_web_sessions",
+      "WildFly Memory Metrics~Accumulated GC Duration"                    => "mw_accumulated_gc_duration",
+      "WildFly Memory Metrics~Heap Max"                                   => "mw_heap_max",
+      "WildFly Memory Metrics~Heap Committed"                             => "mw_heap_committed",
+      "WildFly Memory Metrics~NonHeap Used"                               => "mw_non_heap_used",
+      "WildFly Aggregated Web Metrics~Aggregated Servlet Request Count"   => "mw_aggregated_servlet_request_count",
+      "WildFly Aggregated Web Metrics~Aggregated Active Web Sessions"     => "mw_aggregated_active_web_sessions",
+      "WildFly Aggregated Web Metrics~Aggregated Rejected Web Sessions"   => "mw_aggregated_rejected_web_sessions",
+      "WildFly Threading Metrics~Thread Count"                            => "mw_thread_count",
+      "Server Availability~App Server"                                    => "mw_availability_app_server"
+  }.freeze
+
+  after_initialize do
+    @metrics_capture = ManageIQ::Providers::Hawkular::MiddlewareManager::LiveMetricsCapture.new(self)
+  end
+
+  def metrics_available
+    @metrics_capture.metrics_available
+  end
+
+  def metric(metric, start_time, end_time, interval)
+    @metrics_capture.metric(metric, start_time, end_time, interval)
+  end
+
+  def first_and_last_capture(interval_name = "hourly")
+    captures = [nil, nil]
+    @metrics_capture.metrics_available.each do |metric|
+      capture =  @metrics_capture.first_and_last_capture(metric)
+      if captures[0].nil? || captures[0] > capture[0]
+        captures[0] = capture[0]
+      end
+      if captures[1].nil? || captures[1] < capture[1]
+        captures[1] = capture[1]
+      end
+    end
+    captures
```

Comment:
I think you can simplify this logic with Array#min and Array#max...something like

```ruby
def first_and_last_capture(interval_name = "hourly")
  firsts, lasts = @metrics_capture.metrics_available.collect do |metric|
    @metrics_capture.first_and_last_capture(metric)
  end.transpose
  [firsts.min, lasts.max]
end
```